grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -215,6 +215,73 @@ class InvoiceService {
         invoice.save()
     }
 
+    def submitInvoice(Invoice invoice) {
+        invoice.dateSubmitted = new Date()
+        invoice.save()
+    }
+
+    Invoice generatePrepaymentInvoice(Order order) {
+        if (order.orderItems.any { it.hasInvoices } || order.orderAdjustments.any { it.hasInvoice }) {
+            throw new Exception("Some order items or order adjustments for this order already have been invoiced")
+        }
+
+        Invoice invoice = createFromOrder(order)
+        invoice.invoiceType = InvoiceType.findByCode(InvoiceTypeCode.PREPAYMENT_INVOICE)
+        createOrUpdateVendorInvoiceNumber(invoice, order.orderNumber + Constants.PREPAYMENT_INVOICE_SUFFIX)
+
+        if (order.orderItems.size() + order.orderAdjustments.size() == 0) {
```

Comment:
Careful here. I think Hibernate/Grails will initialize the orderItems and orderAdjustments with an empty list, but for unit tests it might not so you'll need to add null safety. 

Plus I would suggest using something more readable like this.
```
if (!order?.orderItems?.empty && !order?.orderAdjustments?.empty)
```

Reply:
Thanks